automation_tools/__init__.py

```diff
@@ -370,6 +370,45 @@ def setup_fake_manifest_certificate(certificate_url=None):
     manage_daemon('restart', 'tomcat6' if distro_info()[1] <= 6 else 'tomcat')
 
 
+def setup_firewall():
+    """Setup firewall rules that Satellite 6 needs to work properly"""
+    ports = (
+        # Port 443 for HTTPS (secure WWW) must be open for incoming
+        # connections.
+        443,
+        # Port 5671 must be open for SSL communication with managed systems.
+        5671,
+        # Port 80 for HTTP (WWW) must be open to download the bootstrap files.
+        80,
+        # Port 8140 must be open for incoming Puppet connections with the
+        # managed systems.
+        8140,
+        # Port 9090 must be open for Foreman Smart Proxy connections with the
+        # managed systems.
+        9090,
+        # Port 22 must be open for connections via ssh
+        22,
+        # Port 5000 must be open for Docker registry communication.
+        5000,
+    )
+
+    for port in ports:
+        rule_exists = run(
+            r'iptables -nL | grep -E "^ACCEPT\s+tcp.*{0}"'.format(port),
```

Comment:
It wouldn't hurt to specify that you're working with the INPUT iptables chain, like so:

    iptables -nL INPUT

This is because there are also FORWARD and OUTPUT tables, and you want to ignore them.

Also, remember that GNU grep has a very unusual syntax for specifying character classes. Whereas perl and many other regex engines specify the "whitespace" character class with `\s`, GNU grep uses `[[:space:]]`. As a result, the regex you've written should fail. This expression succeeds on my home server:

    "^ACCEPT[[:space:]]\+tcp.*443"

In addition, I recommend adding `$` to the end of the regex, as it is possible that an IP address could match what you're searching for:

    ACCEPT     tcp  --  0.0.0.0/0            192.168.0.80/0            tcp dpt:443

Finally, in GNU grep, there is no difference between basic and extended regular expressions, so the `-E` flag can be dropped. If the suggestions above are taken, then the command becomes:

    r'iptables -nL INPUT | grep "^ACCEPT[[:space:]]\+tcp.*{0}$"'.format(port),

I've checked the above against `man grep` on several of our RHEL systems. I've also taken the firewall output from my home server, SCP'd it to one of our RHEL systems, and ran grep against the file. Everything checks out. (There are no firewall rules on the RHEL systems I have right now, and I don't want to mess things up by adding some.)